### PR TITLE
Docs: redesign home page, and add new theme

### PR DESCRIPTION
**Pull request by lauranovich**
This PR
1. Upgrades the Sphinx theme to 1.o
2. Fixes issues in compiling
3. changes the Home Page to the more modern design
4. Includes the new Scylla-Monitoring Stack mascot

### CODE REVIEW

**Comment by lauranovich** [comment]
@dgarcia360 the tiling is not working - please take a look and comment

File: docs/source/index.rst [hunk 1]
```diff
@@ -1,6 +1,88 @@
...
+  <div class="topics-grid grid-container full">
+
+  <div class="grid-x grid-margin-x">
+
+.. topic-box::
+    :title: Download and Install
+    :icon: fa fa-download
+    :link: /install/
+    :anchor: Get Started
+
```
Comment (davidgarcia-alation) [inline docs/source/index.rst:33]:
```suggestion
    :class: large-6

````

Comment (davidgarcia-alation) [reply docs/source/index.rst:33]: @lauranovich Please add this option to every other topic box.

File: docs/Makefile [hunk 2]
```diff
@@ -75,4 +75,10 @@ multiversion: setup
```
Comment (davidgarcia-alation) [inline docs/Makefile:75]: @lauranovich After the line 12 (Makefile), please add the following new line:

```
TESTSPHINXOPTS  = $(ALLSPHINXOPTS) -W --keep-going
````

I'm updating the upgrade guide to reflect this change.

File: docs/source/index.rst (hunk 1 above)
Comment (dgarcia360) [inline docs/source/index.rst:34]:
```suggestion
    :class: large-6

```

@lauranovich Please add this option to every other topic box.

File: docs/Makefile [hunk 3]
```diff
@@ -75,4 +75,10 @@ multiversion: setup
...
 .PHONY: multiversionpreview
 multiversionpreview: multiversion
-	$(POETRY) run python3 -m http.server 5500 --directory $(BUILDDIR)/dirhtml 
+	$(POETRY) run python3 -m http.server 5500 --directory $(BUILDDIR)/dirhtml
+
+.PHONY: test
+test: setup
+	$(SPHINXBUILD) -b dirhtml $(TESTSPHINXOPTS) $(BUILDDIR)/dirhtml
+	@echo
+	@echo "Build finished. The HTML pages are in $(BUILDDIR)/dirhtml."
```
Comment (dgarcia360) [inline docs/Makefile]: @lauranovich After the line 12 (Makefile), please add the following new line:

```
TESTSPHINXOPTS  = $(ALLSPHINXOPTS) -W --keep-going
```

I'm updating the upgrade guide to reflect this change.

File: docs/source/index.rst [hunk 4]
```diff
@@ -1,6 +1,88 @@
...
-Scylla Monitoring Stack
-========================
+.. title:: Scylla Monitoring Stack
+
+.. raw:: html
+
+  <div class="landing landing--floating">
+
+
+.. hero-box::
```
Comment (davidgarcia-alation) [inline docs/source/index.rst:8]: The hero box does not look visually appealing due to the amount of text:

image

Instead, you could try to make the description shorter as follows:

**Option A) Keep a short description (recommended)**

image


**Option B)  Remove the advanced description, which is already available here https://monitoring.docs.scylladb.com/master/intro.html**

image

**Option C) Remove where to find old versions of the docs, which is already defined in https://monitoring.docs.scylladb.com/master/intro.html**

image

Comment (dgarcia360) [inline docs/source/index.rst:8]: The hero box does not look visually appealing due to the amount of text:

image

Instead, you could try to make the description shorter as follows:

**Option A) Keep a short description (recommended)**

image


**Option B)  Remove the advanced description, which is already available here https://monitoring.docs.scylladb.com/master/intro.html**

image

**Option C) Remove where to find old versions of the docs, which is already defined in https://monitoring.docs.scylladb.com/master/intro.html**

image

File: docs/source/index.rst [hunk 5]
```diff
@@ -1,6 +1,88 @@
...
+.. raw:: html
+
+  <div class="topics-grid grid-container full">
+
+  <div class="grid-x grid-margin-x">
+
+.. topic-box::
+    :title: Download and Install
+    :icon: fa fa-download
+    :link: /install/
```
Comment (dgarcia360) [inline docs/source/index.rst]:
``` suggestion
    :link: install
```

@lauranovich Please do not add leading and trailing "/" (applies to all topic boxes in index.rst).

File: docs/source/index.rst (hunk 1 above)
Comment (lauranovich) [reply docs/source/index.rst:34]: can we add this to the directions?

Comment (lauranovich) [reply docs/source/index.rst:34]: every cell needs it not every other

**Comment by lauranovich** [comment]
@dgarcia360 - I think this is now ready

File: docs/source/index.rst [hunk 6]
```diff
@@ -1,6 +1,95 @@
...
+.. hero-box::
+  :title: Scylla Monitoring Stack Documentation
+  :image: /_static/img/mascots/scylla-monitor.svg
+  :button_icon: fa fa-github
+  :button_url: https://github.com/scylladb/scylla-monitoring
+  :button_text: Scylla Monitoring Stack Project
+
+  Welcome to Scylla Monitoring Stack Documentation!
+  Scylla Monitoring Stack is a full stack for Scylla monitoring and alerting.
+  Older versions of Scylla Monitoring Stack Documentation are available `here <https://docs.scylladb.com/operating-scylla/monitoring/>`_.
```
Comment (tzach) [inline docs/source/index.rst]:
```suggestion
  A full stack for Scylla monitoring and alerting
```

Comment (tzach) [reply docs/source/index.rst]: Making the subtitle shorter and to the point. No one care about the old version any more

**Comment by tzach** [comment]
Remove inner page ToC, for example `/install/monitoring_stack/` 
image

the left menu replace it now

**Comment by lauranovich** [comment]
@dgarcia360 please look at the comments inside this PR
some items you asked me to do didn't work as described or were not included in the documentation
we need to change that, please!

**Comment by tzach** [comment]
Link to extrabl resource need the external link icons, including in the left menu.
See GitHub Project, Scylla University Lesson below
image

**Comment by lauranovich** [comment]
@tzach I think we're good to go the External Link not being flagged @dgarcia360  - this is a regression. We need this fixed.

**Comment by tzach** [comment]
LGTM

**Comment by tzach** [comment]
I see 
```
404 GET /_static/expertrec.js
```
errors from the site

File: docs/Makefile (hunk 3 above)
Comment (dgarcia360) [reply docs/Makefile]: @lauranovich Could you please add the new line after the line 13 (Under "# Internal variables.", after "ALLSPHINXOPTS ...")? It's a aesthetic change to have a Makefile more organized. Now, ``TESTSPHINXOPTS  = $(ALLSPHINXOPTS) -W --keep-going`` is at the end of the file.

**Comment by lauranovich** [comment]
@tzach - the sidebar issue is being handled by the theme issue @dgarcia360 mentioned. ( scylladb/sphinx-scylladb-theme#227) so I think we can merge. @dgarcia360 any other changes?

**Comment by tzach** [comment]
Waiting for a solution (or understanding) of the new 404 issue above

File: docs/Makefile (hunk 3 above)
Comment (lauranovich) [reply docs/Makefile]: done

**Comment by tzach** [comment]
> 404 GET /_static/expertrec.js

Issue source is the following HTML code
```
<script src="_static/expertrec.js"></script>
```
Look like the new theme dropped the expertrec.js which do exist in older version

**Comment by tzach** [comment]
https://github.com/scylladb/sphinx-scylladb-theme/issues/228

**Comment by lauranovich** [comment]
Linkcheck passes on the PR, fixed some old broken links

File: docs/source/install/monitoring_stack.rst [hunk 7]
```diff
@@ -10,12 +10,12 @@ For evaluation, you can run Scylla Monitoring Stack on any server (or laptop) th
...
 .. include:: min-prod-hw.rst
 
-.. _`Scylla Prometheus API`: monitoring_apis#prometheus
+.. _`Scylla Prometheus API`: /reference/monitoring_apis#prometheus
 
 Prerequisites
 -------------
 
-* Follow the Installation Guide and install `docker`_ on the Scylla Monitoring Stack Server. This server can be the same server that is running Scylla Manager. Alternatively, you can `Deploy Scylla Monitoring Stack Without Docker <monitor_without_docker>`_ .
+* Follow the Installation Guide and install `docker`_ on the Scylla Monitoring Stack Server. This server can be the same server that is running Scylla Manager. Alternatively, you can `Deploy Scylla Monitoring Stack Without Docker </install/monitor_without_docker>`_ .
```
Comment (dgarcia360) [inline docs/source/install/monitoring_stack.rst]: Internal links are constructed with the ":doc:" directive. For more information, see [Links](https://sphinx-theme.scylladb.com/stable/examples/links.html).

```suggestion
* Follow the Installation Guide and install `docker`_ on the Scylla Monitoring Stack Server. This server can be the same server that is running Scylla Manager. Alternatively, you can :doc:`Deploy Scylla Monitoring Stack Without Docker <monitor_without_docker>`.
```

File: docs/source/install/monitoring_stack.rst [hunk 8]
```diff
@@ -10,12 +10,12 @@ For evaluation, you can run Scylla Monitoring Stack on any server (or laptop) th
 
 .. include:: min-prod-hw.rst
 
-.. _`Scylla Prometheus API`: monitoring_apis#prometheus
+.. _`Scylla Prometheus API`: /reference/monitoring_apis#prometheus
```
Comment (dgarcia360) [inline docs/source/install/monitoring_stack.rst:13]: Relative URLs should not start with / (slash).

```suggestion
```` 

Suggestion: Delete this line, and use the ``:doc:`` directive instead.

Comment (dgarcia360) [inline docs/source/install/monitoring_stack.rst]: Relative URLs should not start with / (slash).

**Possible solution:** Delete this line, and in line 5, use a cross reference link instead:

```
* This is an example of an :ref:`Internal Link <link-format-examples>`. This is an internal cross reference. It requires a bookmark.
```

For more information, see [Links](https://sphinx-theme.scylladb.com/stable/examples/links.html).

**Comment by dgarcia360** [comment]
@tzach  Expertrec.js is now inlined in the theme's layout:

- Old design: https://github.com/scylladb/sphinx-scylladb-theme/blob/branch-0.1/sphinx_scylladb_theme/static/expertrec.js
- New design: https://github.com/scylladb/sphinx-scylladb-theme/blob/d2a771b3120a73553b5d560e088e8eb7c6fcf659/sphinx_scylladb_theme/layout.html#L139-L149

@lauranovich @tzach Please remove or comment out the following lines in scylla-monitoring [conf.py](https://github.com/scylladb/scylla-monitoring/blob/master/docs/source/conf.py#L135-L136):

```
# Add ExpertRec Search
html_js_files = ['expertrec.js']
```

I'm adding this detail in the upgrade guides.

**Comment by dgarcia360** [comment]
> some items you asked me to do didn't work as described or were not included in the documentation
we need to change that, please!

@lauranovich I'm updating the docs with in-detail reference documentation for options of the components "topic-box", "hero-box", and "panel-box".

> External Link not being flagged @dgarcia360 - this is a regression

Thanks, I didn't spot this bug yet. I'll fix it with https://github.com/scylladb/sphinx-scylladb-theme/issues/227

**Pull request merged**